Auto update failing for p11-kit

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [p11-kit](https://github.com/termux/termux-packages/blob/cef9676fdabf7c9b47a5f457d202e0223f57e2f8/packages/p11-kit) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating p11-kit [Current version: 0.25.8]
INFO: package being updated to 0.25.9.
Downloading https://github.com/p11-glue/p11-kit/releases/download/0.25.9/p11-kit-0.25.9.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  518k  100  518k    0     0   997k      0 --:--:-- --:--:-- --:--:--  997k

You are about to commit these changes:

--------------------
diff --git a/packages/p11-kit/build.sh b/packages/p11-kit/build.sh
index f6818bab11..e49cd5acf8 100644
--- a/packages/p11-kit/build.sh
+++ b/packages/p11-kit/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE="https://p11-glue.github.io/p11-glue/p11-kit.html"
 TERMUX_PKG_DESCRIPTION="Provides a way to load and enumerate PKCS#11 modules"
 TERMUX_PKG_LICENSE="BSD 3-Clause"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="0.25.8"
+TERMUX_PKG_VERSION="0.25.9"
 TERMUX_PKG_SRCURL="https://github.com/p11-glue/p11-kit/releases/download/$TERMUX_PKG_VERSION/p11-kit-$TERMUX_PKG_VERSION.tar.xz"
-TERMUX_PKG_SHA256=2fd4073ee2a47edafaae2c8affa2bcca64e0697f8881f68f580801ef43cab0ce
+TERMUX_PKG_SHA256=f6512a10b2dcf2651cfd57dd767a36c6e44494ab37724c10d4304fe9f0a36497
 TERMUX_PKG_DEPENDS="libffi, libtasn1"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--without-trust-paths --disable-static"
 TERMUX_PKG_AUTO_UPDATE=true
--------------------

bump(main/p11-kit): 0.25.9


Not committing to Git!

INFO: Trying to build package.
INFO: Package p11-kit exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building p11-kit for arch aarch64...
::group::INFO: Fetching repo metadata
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100 13134  100 13134    0     0   167k      0 --:--:-- --:--:-- --:--:--  168k
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100   833  100   833    0     0   2815      0 --:--:-- --:--:-- --:--:--  2823
[packages-cf.termux.dev-apt-termux-main]: gpg: Signature made Thu 18 Sep 2025 06:26:11 PM UTC
[packages-cf.termux.dev-apt-termux-main]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-main]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-main]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100 13320  100 13320    0     0  58158      0 --:--:-- --:--:-- --:--:-- 58421
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100   833  100   833    0     0   3318      0 --:--:-- --:--:-- --:--:--  3318
[packages-cf.termux.dev-apt-termux-root]: 100   833  100   833    0     0   3317      0 --:--:-- --:--:-- --:--:--  3305
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100 13120  100 13120    0     0  44745      0 --:--:-- --:--:-- --:--:-- 44778
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: gpg: Signature made Sat 13 Sep 2025 05:36:01 AM UTC
[packages-cf.termux.dev-apt-termux-root]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-root]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100   833  100   833    0     0   3511      0 --:--:-- --:--:-- --:--:--  3500
[packages-cf.termux.dev-apt-termux-x11]: gpg: Signature made Thu 18 Sep 2025 09:32:56 AM UTC
[packages-cf.termux.dev-apt-termux-x11]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-x11]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
::endgroup::
Downloading dependency libffi@3.4.7-1 if necessary...
Found libffi in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/libf/libffi/libffi_3.4.7-1_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 30672  100 30672    0     0   514k      0 --:--:-- --:--:-- --:--:--  516k
extracting libffi to /home/builder/.termux-build/_cache-aarch64...
Downloading dependency libtasn1@4.20.0-1 if necessary...
Found libtasn1 in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/libt/libtasn1/libtasn1_4.20.0-1_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   99k  100   99k    0     0  1234k      0 --:--:-- --:--:-- --:--:-- 1241k
extracting libtasn1 to /home/builder/.termux-build/_cache-aarch64...
Downloading p11-kit@0.25.9 source from 'https://github.com/p11-glue/p11-kit/releases/download/0.25.9/p11-kit-0.25.9.tar.xz' if necessary...
Downloading https://github.com/p11-glue/p11-kit/releases/download/0.25.9/p11-kit-0.25.9.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  518k  100  518k    0     0  4456k      0 --:--:-- --:--:-- --:--:-- 4456k
autopoint: using AM_GNU_GETTEXT_REQUIRE_VERSION instead of AM_GNU_GETTEXT_VERSION
Copying file ABOUT-NLS
Copying file build/litter/config.rpath
Copying file build/m4/gettext.m4
Copying file build/m4/host-cpu-c-abi.m4
Copying file build/m4/iconv.m4
Copying file build/m4/intlmacosx.m4
Copying file build/m4/lib-ld.m4
Copying file build/m4/lib-link.m4
Copying file build/m4/lib-prefix.m4
Copying file build/m4/nls.m4
Copying file build/m4/po.m4
Copying file build/m4/progtest.m4
Copying file po/Makefile.in.in
Copying file po/Makevars.template
Copying file po/Rules-quot
Copying file po/boldquot.sed
Copying file po/en@boldquot.header
Copying file po/en@quot.header
Copying file po/insert-header.sin
Copying file po/quot.sed
Copying file po/remove-potcdate.sin
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build/litter'.
libtoolize: copying file 'build/litter/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'build/m4'.
libtoolize: copying file 'build/m4/libtool.m4'
libtoolize: copying file 'build/m4/ltoptions.m4'
libtoolize: copying file 'build/m4/ltsugar.m4'
libtoolize: copying file 'build/m4/ltversion.m4'
libtoolize: copying file 'build/m4/lt~obsolete.m4'
configure.ac:541: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:541: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:541: the top level
configure.ac:39: installing 'build/litter/compile'
configure.ac:42: installing 'build/litter/config.guess'
configure.ac:42: installing 'build/litter/config.sub'
configure.ac:31: installing 'build/litter/install-sh'
configure.ac:31: installing 'build/litter/missing'
configure.ac:30: installing 'build/litter/tap-driver.sh'
Makefile.am:48: warning: noinst_SCRIPTS multiply defined in condition TRUE ...
Makefile.am:41: ... 'noinst_SCRIPTS' previously defined here
Makefile.am: installing 'build/litter/depcomp'
parallel-tests: installing 'build/litter/test-driver'
configure: WARNING: unrecognized options: --disable-rpath-hack
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for aarch64-linux-android-strip... llvm-strip
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether build environment is sane... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for aarch64-linux-android-gcc... aarch64-linux-android-clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether aarch64-linux-android-clang accepts -g... yes
checking for aarch64-linux-android-clang option to enable C11 features... none needed
checking whether aarch64-linux-android-clang understands -c and -o together... yes
checking dependency style of aarch64-linux-android-clang... none
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking build system type... x86_64-pc-linux-gnu
checking host system type... aarch64-unknown-linux-android
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by aarch64-linux-android-clang... ld.lld
checking if the linker (ld.lld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... llvm-nm
checking the name lister (llvm-nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 3145728
checking how to convert x86_64-pc-linux-gnu file names to aarch64-unknown-linux-android format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld.lld option to reload object files... -r
checking for aarch64-linux-android-file... no
checking for file... file
configure: WARNING: using cross tools not prefixed with host triplet
checking for aarch64-linux-android-objdump... llvm-objdump
checking how to recognize dependent libraries... pass_all
checking for aarch64-linux-android-dlltool... no
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for aarch64-linux-android-ar... llvm-ar
checking for archiver @FILE support... @
checking for aarch64-linux-android-strip... (cached) llvm-strip
checking for aarch64-linux-android-ranlib... llvm-ranlib
checking command to parse llvm-nm output from aarch64-linux-android-clang object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for aarch64-linux-android-mt... no
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if aarch64-linux-android-clang supports -fno-rtti -fno-exceptions... yes
checking for aarch64-linux-android-clang option to produce PIC... -fPIC -DPIC
checking if aarch64-linux-android-clang PIC flag -fPIC -DPIC works... yes
checking if aarch64-linux-android-clang static flag -static works... yes
checking if aarch64-linux-android-clang supports -c -o file.o... yes
checking if aarch64-linux-android-clang supports -c -o file.o... (cached) yes
checking whether the aarch64-linux-android-clang linker (ld.lld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... Android linker
checking how to hardcode library paths into programs... immediate
checking for shl_load... no
checking for shl_load in -ldld... no
checking for dlopen... yes
checking whether a program can dlopen itself... cross
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking if LD -Wl,--version-script works... no
checking whether linker understands -z nodelete... yes
checking for aarch64-linux-android-gcc... (cached) aarch64-linux-android-clang
checking whether the compiler supports GNU C... (cached) yes
checking whether aarch64-linux-android-clang accepts -g... (cached) yes
checking for aarch64-linux-android-clang option to enable C11 features... (cached) none needed
checking whether aarch64-linux-android-clang understands -c and -o together... (cached) yes
checking dependency style of aarch64-linux-android-clang... (cached) none
checking how to run the C preprocessor... aarch64-linux-android-cpp
checking pkg-config is at least version 0.9.0... yes
checking for a Python interpreter with version >= 3.6... python
checking for python... /usr/bin/python
checking for python version... 3.12
checking for python platform... linux
checking for GNU default python prefix... ${prefix}
checking for GNU default python exec_prefix... ${exec_prefix}
checking for python script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.12/site-packages
checking for python extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib/python3.12/site-packages
checking whether NLS is requested... no
checking for msgfmt... /usr/bin/msgfmt
checking for gmsgfmt... /usr/bin/msgfmt
checking for xgettext... /usr/bin/xgettext
checking for msgmerge... /usr/bin/msgmerge
checking for ld... ld.lld
checking if the linker (ld.lld) is GNU ld... yes
checking for shared library run path origin... done
checking 32-bit host C ABI... no
checking for ELF binary format... yes
checking for the common suffixes of directories in the library search path... lib,lib,lib64
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyPreferredLanguages... no
checking whether to use NLS... no
checking for win32... no
checking whether byte ordering is bigendian... (cached) no
checking for _Bool... yes
checking for stdbool.h that conforms to C99... yes
checking size of unsigned long... 8
checking for pthread_create... yes
checking for nanosleep... yes
checking for library containing dlopen... none required
checking for library containing gethostbyname... none required
checking for library containing connect... none required
checking for locale.h... yes
checking for locale_t... yes
checking for newlocale... yes
checking for strerror_l... yes
checking for sys/resource.h... yes
checking for sys/un.h... yes
checking for ucred.h... no
checking for struct dirent.d_type... yes
checking for getprogname... yes
checking for getexecname... no
checking for basename... yes
checking for mkstemp... yes
checking for mkdtemp... yes
checking for getresuid... yes
checking for secure_getenv... no
checking for getauxval... yes
checking for strnstr... no
checking for memdup... no
checking for strndup... yes
checking for reallocarray... no
checking for aarch64-linux-android-clang options needed to detect all undeclared functions... none needed
checking whether reallocarray is declared... no
checking for fdwalk... no
checking for setenv... yes
checking for getpeereid... no
checking for getpeerucred... no
checking for issetugid... no
checking for isatty... yes
checking for readpassphrase... no
checking for strerror_r... yes
checking for thread-local storage class... __thread
checking for gmtime_r... yes
checking whether program_invocation_short_name is declared... no
checking whether program_invocation_short_name is available... no
checking whether __progname is declared... no
checking for sighandler_t... yes
checking for sig_t... yes
checking for __sighandler_t... yes
checking whether vsock support is available... yes
checking for asprintf... yes
checking for vasprintf... yes
checking for vsnprintf... yes
checking whether asprintf is declared... yes
checking whether vasprintf is declared... yes
checking for dgettext in -lintl... no
checking for libtasn1 >= 2.3... yes
checking for asn1Parser... no
checking for libffi >= 3.0.0... yes
checking if trust module is enabled... yes
checking for trust module paths... disabled
checking for gtkdoc-check... /usr/bin/gtkdoc-check
checking for gtkdoc-rebase... /usr/bin/gtkdoc-rebase
checking for gtkdoc-mkpdf... /usr/bin/gtkdoc-mkpdf
checking for gtkdoc-scan... /usr/bin/gtkdoc-scan
checking for xsltproc... /usr/bin/xsltproc
checking whether to build documentation... no
checking for debug mode... default (-g, debug output)
checking for more warnings... checking whether gcc understands -Wmissing-include-dirs... yes
checking whether gcc understands -Wundef... yes
checking build strict... no
checking whether to build with gcov testing... no
checking that generated files are newer than configure... done
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating common/test-init.sh
config.status: creating doc/Makefile
config.status: creating doc/manual/Makefile
config.status: creating po/Makefile.in
config.status: creating p11-kit/p11-kit-1.pc
config.status: creating p11-kit/pkcs11.conf.example
config.status: creating p11-kit/version.h
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing po-directories commands
config.status: creating po/POTFILES
config.status: creating po/Makefile
configure: WARNING: unrecognized options: --disable-rpath-hack
configure: build options:

    Host:                            aarch64-unknown-linux-android
    Debug build:                     default (-g, debug output)
    Strict build:                    no
    Build documentation:             no (no manual or reference)
    System global config:            ${prefix}/etc/pkcs11/pkcs11.conf
    System module config directory:  ${prefix}/etc/pkcs11/modules
    Package module config directory: ${pkgdatadir}/modules
    User global config:              ~/.config/pkcs11/pkcs11.conf
    User module config directory:    ~/.config/pkcs11/modules
    Load relative module paths from: /data/data/com.termux/files/usr/lib/pkcs11

    With libtasn1 dependency:        yes
    With libffi:                     yes
    With hash implementation:        internal
    With systemd:                    no

    Build trust module:              yes
    Trust module paths:              

    RPC protocol versions:           0 upto 2


/bin/bash: line 1: -o: command not found
/bin/bash: line 1: -o: command not found
  GEN      common/basic.asn.h
  GEN      common/openssl.asn.h
make: *** [Makefile:7391: common/basic.asn.h] Error 127
make: *** Waiting for unfinished jobs....
make: *** [Makefile:7391: common/openssl.asn.h] Error 127
/bin/bash: line 1: -o: command not found
  GEN      common/pkix.asn.h
  GEN      /home/builder/.termux-build/p11-kit/src/subprojects/pkcs11-json/generated/pkcs11.json
make: *** [Makefile:7391: common/pkix.asn.h] Error 127
/home/builder/.termux-build/p11-kit/src/build/litter/missing: line 81: castxml: command not found
WARNING: 'castxml' is missing on your system.
         You might have modified some files without having the proper
         tools for further handling them.  Check the 'README' file, it
         often tells you about the needed prerequisites for installing
         this package.  You may also peek at any GNU archive site, in
         case some other package contains this missing 'castxml' program.
make: *** [Makefile:7482: /home/builder/.termux-build/p11-kit/src/subprojects/pkcs11-json/generated/pkcs11.json] Error 127
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2025-09-18 18:30:36 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/17837284650">17837284650</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/p11-kit/0001-workaround-asn1Parser-for-cross-compile.diff
+++ b/packages/p11-kit/0001-workaround-asn1Parser-for-cross-compile.diff
@@ -1,0 +1,38 @@
+--- a/common/meson.build
++++ b/common/meson.build
+@@ -88,21 +88,21 @@
+     'basic.asn.h',
+     output: 'basic.asn.h',
+     input: 'basic.asn',
+-    command: [asn1Parser, '-o', '@OUTPUT@', '@INPUT@'],
++    command: ['termux-proot-run', asn1Parser, '-o', '@OUTPUT@', '@INPUT@'],
+   )
+
+   pkix_asn_h = custom_target(
+     'pkix.asn.h',
+     output: 'pkix.asn.h',
+     input: 'pkix.asn',
+-    command: [asn1Parser, '-o', '@OUTPUT@', '@INPUT@'],
++    command: ['termux-proot-run', asn1Parser, '-o', '@OUTPUT@', '@INPUT@'],
+   )
+
+   openssl_asn_h = custom_target(
+     'openssl.asn.h',
+     output: 'openssl.asn.h',
+     input: 'openssl.asn',
+-    command: [asn1Parser, '-o', '@OUTPUT@', '@INPUT@'],
++    command: ['termux-proot-run', asn1Parser, '-o', '@OUTPUT@', '@INPUT@'],
+   )
+
+   asn_h_dep = declare_dependency(
+--- a/meson.build
++++ b/meson.build
+@@ -440,7 +440,7 @@
+ libtasn1 = dependency('libtasn1', version: '>= 2.3',
+                       required: get_option('trust_module'))
+ if libtasn1.found()
+-  asn1Parser = find_program('asn1Parser', required: get_option('trust_module'))
++  asn1Parser = find_program('asn1Parser', '@TERMUX_PREFIX@/bin/asn1Parser', required: get_option('trust_module'))
+   if asn1Parser.found()
+     conf.set('WITH_ASN1', 1)
+     libtasn1_deps += libtasn1

--- a/packages/p11-kit/build.sh
+++ b/packages/p11-kit/build.sh
@@ -2,12 +2,24 @@ TERMUX_PKG_HOMEPAGE="https://p11-glue.github.io/p11-glue/p11-kit.html"
 TERMUX_PKG_DESCRIPTION="Provides a way to load and enumerate PKCS#11 modules"
 TERMUX_PKG_LICENSE="BSD 3-Clause"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="0.25.8"
+TERMUX_PKG_VERSION="0.25.9"
 TERMUX_PKG_SRCURL="https://github.com/p11-glue/p11-kit/releases/download/$TERMUX_PKG_VERSION/p11-kit-$TERMUX_PKG_VERSION.tar.xz"
-TERMUX_PKG_SHA256=2fd4073ee2a47edafaae2c8affa2bcca64e0697f8881f68f580801ef43cab0ce
+TERMUX_PKG_SHA256=f6512a10b2dcf2651cfd57dd767a36c6e44494ab37724c10d4304fe9f0a36497
 TERMUX_PKG_DEPENDS="libffi, libtasn1"
-TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--without-trust-paths --disable-static"
+TERMUX_PKG_BUILD_DEPENDS="aosp-libs, bash-completion"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-Dtrust_module=enabled
+"
 TERMUX_PKG_AUTO_UPDATE=true
+
+termux_step_pre_configure() {
+	if [[ "$TERMUX_ON_DEVICE_BUILD" == "false" ]]; then
+		termux_setup_proot
+		sed \
+		-e "s%\@TERMUX_PREFIX\@%${TERMUX_PREFIX}%g" \
+		"$TERMUX_PKG_BUILDER_DIR"/0001-workaround-asn1Parser-for-cross-compile.diff | patch -p1
+	fi
+}
 
 termux_step_post_get_source() {
 	# Do not forget to bump revision of reverse dependencies and rebuild them
@@ -23,8 +35,4 @@ termux_step_post_get_source() {
 	if [ ! "${_LT_CURRENT}" ] || [ "${v}" != "${_SOVERSION}" ]; then
 		termux_error_exit "SOVERSION guard check failed."
 	fi
-}
-
-termux_step_pre_configure() {
-	autoreconf -fi
 }

